library/src/contraction/contraction_solution_impl.hpp

```diff
@@ -232,10 +389,723 @@ namespace hiptensor
+                                            std::vector<std::size_t> const& a_ms_ns_lengths,
+                                            std::vector<std::size_t> const& a_ms_ks_strides,
+                                            std::vector<std::size_t> const& b_ns_ks_lengths,
+                                            std::vector<std::size_t> const& b_ns_ks_strides,
+                                            std::vector<std::size_t> const& ds_ms_ns_lengths,
+                                            std::vector<std::size_t> const& ds_ms_ns_strides,
+                                            std::vector<std::size_t> const& e_ms_ns_lengths,
+                                            std::vector<std::size_t> const& e_ms_ns_strides,
+                                            void*                           workspacePtr,
+                                            StreamConfig const& streamConfig /*= StreamConfig{}*/) override
+        {
+            if(!initArgs(alpha,
+                        A,
+                        B,
+                        beta,
+                        D,
+                        E,
+                        a_ms_ns_lengths,
+                        a_ms_ks_strides,
+                        b_ns_ks_lengths,
+                        b_ns_ks_strides,
+                        ds_ms_ns_lengths,
+                        ds_ms_ns_strides,
+                        e_ms_ns_lengths,
+                        e_ms_ns_strides,
+                        workspacePtr))
+            {
+    #if !NDEBUG
+                std::cout << kernelName() << " does not support this problem" << std::endl;
+    #endif // !NDEBUG
+                return -1.0f;
+            }
+
+            return mInvokerPtr->Run(mArgPtr.get(), streamConfig);
+        }
+
+    };
+
+    //-------------------------------------------------------------------------------------------------//
+    // hipFloatComplex
+
+    template <typename DeviceOp, typename DataType>
+    class ContractionSolutionImpl<
+        DeviceOp,
+        DataType,
+        std::enable_if_t<(std::is_same<DataType, hipFloatComplex>{} ||
+                          std::is_same<DataType, hipDoubleComplex>{}) &&
+                          std::is_same_v<typename MetaTraits<DeviceOp>::CDEOp,
+                                        ck::tensor_operation::element_wise::Bilinear>>>
+        : public ContractionSolution
+    {
+    protected :
+        std::vector<std::unique_ptr<ck::tensor_operation::device::BaseArgument>> mArgPtr;
+        std::vector<std::unique_ptr<ck::tensor_operation::device::BaseInvoker>>  mInvokerPtr;
+
+        using elementType = typename std::conditional<std::is_same<DataType, hipFloatComplex>::value,
+                                                      float, double>::type;
+
+        std::unique_ptr<elementType, DeviceDeleter> A_d_real;
+        std::unique_ptr<elementType, DeviceDeleter> A_d_imag;
+        std::unique_ptr<elementType, DeviceDeleter> B_d_real;
```

Comment:
You may want to define an alias for the device allocations
```
template<typename DataT>
using DevicePtrT = std::unique_ptr<DataT, DeviceDeleter>;

DevicePtrT mAReal;
DevicePtrT mAImag;
...
```

This will help to cut down on repetition